### PR TITLE
feat: collect rum data

### LINKED ISSUE #24: Integrate RUM data collection
In order to be able to do some reporting (especially validate usage increase), we need to start collecting RUM data.

### PR DESCRIPTION
**Pull request by kptdobe**
Fix #24 

first attempt to collect some data:
- login attempts, success and failures (including token validation)
- client connect to channel
- client sends message

I have also set the sampling rate to 1 for now to collect everything and see if this is useful (we do not have a lot of traffic...)

### CODE REVIEW

File: src/AuthProvider.tsx [hunk 1]
```diff
@@ -23,25 +24,31 @@ export const AuthProvider: React.FC<PropsWithChildren> = ({children}) => {
       login: async (email: string) => {
         try {
           console.log(`Trying to login with email: ${email}`);
+          sampleRUM('chat:login', { source: 'attempt' });
```
Comment (trieloff) [inline src/AuthProvider.tsx]: Our `source` values look like CSS selectors, so either
```suggestion
          sampleRUM('chat:login', { source: '.attempt' });
```

or 
```suggestion
          sampleRUM('chat:login', { source: '#attempt' });
```

would work better

File: src/AuthProvider.tsx [hunk 2]
```diff
@@ -23,25 +24,31 @@ export const AuthProvider: React.FC<PropsWithChildren> = ({children}) => {
...
           setState({token: state.token, loading: true});
           const token = await magic.auth.loginWithMagicLink({email});
           setState({token, loading: false});
           console.log(`Logged in with email: ${email}`, JSON.stringify(token));
+          sampleRUM('chat:login', { source: 'success' });
         } catch (e) {
           console.error(`Failed to login with email: ${email}`, e);
           setState({token: state.token, loading: false});
+          sampleRUM('chat:login', { source: 'failed' });
```
Comment (trieloff) [inline src/AuthProvider.tsx]:
```suggestion
          sampleRUM('chat:login-failed', { source: 'failed' });
```

This is a different, distinct checkpoint

File: src/AuthProvider.tsx [hunk 3]
```diff
@@ -23,25 +24,31 @@ export const AuthProvider: React.FC<PropsWithChildren> = ({children}) => {
...
+          sampleRUM('chat:logout', { source: 'attempt' });
           setState({token: state.token, loading: true})
           await magic.user.logout();
           setState({token: null, loading: false})
           console.log(`Logged out`);
+          sampleRUM('chat:logout', { source: 'success' });
         } catch (e) {
           console.error(`Failed to logout`, e);
           setState({token: state.token, loading: false});
+          sampleRUM('chat:logout', { source: 'failed' });
```
Comment (trieloff) [inline src/AuthProvider.tsx]:
```suggestion
          sampleRUM('chat:logout-failed', { source: 'failed' });
```

File: src/ChatClient.ts [hunk 4]
```diff
@@ -62,6 +63,7 @@ export class ChatClient {
         this.client.on("disconnect", () => { this.fireStatusChange(ConnectionStatus.DISCONNECTED) });
         this.client.on("error", (error) => {
             console.log(`Error: ${error}`);
+            sampleRUM('chat:client', { source: 'error', target: error });
```
Comment (trieloff) [inline src/ChatClient.ts]: Don't duplicate information. I'd go with `chat:error` source would be the script or function, target the error message

**Review by trieloff — CHANGES_REQUESTED** [review]

File: src/AuthProvider.tsx (hunk 3 above)
Comment (kptdobe) [reply src/AuthProvider.tsx]: Below you say "don't duplicate information". If I do that, I clearly duplicate info.... If I understand it well, the source might be the function....

**Comment by kptdobe** [comment]
@trieloff ok, 2nd attempt based on feedback.

**Review by trieloff — APPROVED** [review]
Let's try it

**Pull request merged**